tardis/plasma/properties/ion_population.py

```diff
@@ -349,20 +349,4 @@ def calculate(self, phi, partition_function, number_density,
                                                       1].sum(axis=0))
             ion_number_density.ix[2].ix[2].update(helium_population_updated.ix[
                                                       2].ix[0])
-            ion_numbers = ion_number_density.index.get_level_values(1).values
```

Comment:
I think this code block is not required when `electron_densities` are read from a seperate file in `IonNumberDensityHeNLTE` class.

Reply:
@vg3095 well often it isn't - we need to be careful here. It needs to calculate them if they are not given. They will often not be given.

Reply:
@wkerzendorf  
See this [line](https://github.com/vg3095/tardis/blob/ac8e67105baa1667b8ab6f8fe74d8dad492a9777/tardis/plasma/properties/ion_population.py#L321), and [here](https://github.com/vg3095/tardis/blob/ac8e67105baa1667b8ab6f8fe74d8dad492a9777/tardis/plasma/properties/ion_population.py#L350), I have removed this code block  from second link (from `else` part --> if `electron_densities` is present).